src/install/installer.ts

```diff
@@ -52,8 +52,13 @@ async function validateCache(packagePath: string, browsersPath: string, linksDir
     let linkTarget = '';
     try {
       linkTarget = (await fsReadFileAsync(linkPath)).toString();
-      const browsers = JSON.parse((await fsReadFileAsync(path.join(linkTarget, 'browsers.json'))).toString())['browsers'];
+      const browsers = JSON.parse((await fsReadFileAsync(path.join(linkTarget, 'browsers.json'))).toString())['browsers'] as browserPaths.BrowserDescriptor[];
+      // Older versions do not have "download" field. We assume they need all browsers
+      // from the list. For newer ones, only the browsers with "download" are counted.
+      const hasDownloads = browsers.some(b => b.download !== undefined);
       for (const browser of browsers) {
+        if (hasDownloads && !browser.download)
```

Comment:
I had hard time understanding the `hasDownloads` logic.
Maybe the straight-away condition would be simpler to understand?

```js
// Older versions do not have "download" field. We assume they need all browsers
// from the list. So we want to skip all browsers that are explicitly marked as "download: false".
if (browser.download === false)
  continue;
```

Alternatively, maybe we can extract it into a function and reuse here and below in step (3)?

```js
async function readBrowsersToDownload(packagePath) {
  const browsers = JSON.parse((await fsReadFileAsync(path.join(packagePath, 'browsers.json'))).toString())['browsers'] as browserPaths.BrowserDescriptor[];
  // Older versions do not have "download" field. We assume they need all browsers
  // from the list. So we want to skip all browsers that are explicitly marked as "download: false".
  return browsers.filter(browser => browser.download !== false);
}
```

Reply:
Done.